code/script.js

```diff
@@ -244,100 +246,115 @@ const generateBoard = () => {
-  } else if (group === 'other') {
-    // Similar to the one above
+  // Alerts...
+  if (currentQuestion.category === 'accessories') {
+    keep ?
+      alert(`Yes, the person wears ${currentQuestion.attribute}! Keep all that wears ${currentQuestion.attribute}`)
+    :
+      alert(`No, the person doesn't wear ${currentQuestion.attribute}! Remove all that wears ${currentQuestion.attribute}`)
+    
+  } else if (currentQuestion.category === 'hair color') {
+    keep ?
+      alert(`Yes, the person has ${currentQuestion.value} hair! Keep all that have ${currentQuestion.value}`)
+    :
+      alert(`No, the person doesn't have ${currentQuestion.value} hair! Remove all that have ${currentQuestion.value} hair`)
+    
+  } else if (currentQuestion.category === 'eye color') {
+    keep ?
+      alert(`Yes, the person has ${currentQuestion.value} eyes! Keep all that have ${currentQuestion.value} eyes`)
+    :
+      alert(`No, the person doesn't have ${currentQuestion.value} eyes! Remove all that have ${currentQuestion.value}eyes`)
+    
+  } else if (currentQuestion.category === 'other') {
+    keep ?
+      alert(`Yes, the person smokes! Keep all the smokers`)
+    :
+      alert(`No, the person doesn't smoke! Remove all smokers`)
+  }
+  // And here the actual filtering is done
+  if (keep) {
+    charactersInPlay = charactersInPlay.filter((person) => person[currentQuestion.attribute] === currentQuestion.value)
   } else {
-    if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all persons with yellow hair"
-    } else {
-      // alert popup that says something like: "NO, the person doesnt have yellow hair! Remove all persons with yellow hair"
-    }
+    charactersInPlay = charactersInPlay.filter((person) => person[currentQuestion.attribute] !== currentQuestion.value)
   }
-
-  // filter to keep or remove based on the keep variable.
-  /* charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-    or 
-    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value) */
-
-  // Invoke a function to redraw the board with the remaining people.
+  generateBoard()
 }
-
-// when clicking guess, the player first have to confirm that they want to make a guess.
+// User confirms the guess
 const guess = (suspect) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+  const confirmation = confirm(`Are you sure you want to guess ${suspect}?`)
+  if (confirmation) {
+    checkMyGuess(suspect)
+  }
 }
 
-// If you confirm, this function is invoked
 const checkMyGuess = (suspect) => {
-  // 1. Check if the suspect is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+  if (suspect === secret.name) {
+    board.innerHTML = ''
```

Comment:
Is this line necessary?

Reply:
Hey Sofia, thanks for a great code review! My idea for line 343 (and 348) was to empty the game board before displaying the win or lose-message!